src/main/java/net/erv123/shadertoymc/arucas/extension/ShaderExtension.java

```diff
@@ -0,0 +1,138 @@
+		);
+	}
+
+	@NotNull
+	@Override
+	public String getName() {
+		return "ShaderExtension";
+	}
+
+	private String getWorld(Arguments arguments) {
+		return ScriptUtils.getScriptHolder(arguments.getInterpreter()).getWorld().getRegistryKey().getValue().getPath();
+	}
+
+	private Void placeDefault(Arguments arguments) {
+		String block = arguments.nextPrimitive(StringDef.class);
+		int x = arguments.nextPrimitive(NumberDef.class).intValue();
+		int y = arguments.nextPrimitive(NumberDef.class).intValue();
+		int z = arguments.nextPrimitive(NumberDef.class).intValue();
+		ServerCommandSource source = ScriptUtils.getScriptHolder(arguments.getInterpreter());
+		this.place(arguments.getInterpreter(), source.getWorld(), block, x, y, z);
+		return null;
+	}
+
+	private Void placeWithWorld(Arguments arguments) {
+		String block = arguments.nextPrimitive(StringDef.class);
+		int x = arguments.nextPrimitive(NumberDef.class).intValue();
+		int y = arguments.nextPrimitive(NumberDef.class).intValue();
+		int z = arguments.nextPrimitive(NumberDef.class).intValue();
+		String worldString = arguments.nextPrimitive(StringDef.class);
+
+		RegistryKey<World> registry = RuntimeError.wrap(() -> RegistryKey.of(RegistryKeys.WORLD, Identifier.tryParse(worldString)));
+		ServerCommandSource source = ScriptUtils.getScriptHolder(arguments.getInterpreter());
+		ServerWorld world = source.getServer().getWorld(registry);
+
+		if (world == null) {
+			throw new RuntimeError("Failed to get world for: " + worldString);
+		}
+		this.place(arguments.getInterpreter(), world, block, x, y, z);
+		return null;
+	}
+
+	private Void area(Arguments arguments) {
+		int ax = arguments.nextPrimitive(NumberDef.class).intValue();
+		int ay = arguments.nextPrimitive(NumberDef.class).intValue();
+		int az = arguments.nextPrimitive(NumberDef.class).intValue();
+		int bx = arguments.nextPrimitive(NumberDef.class).intValue();
+		int by = arguments.nextPrimitive(NumberDef.class).intValue();
+		int bz = arguments.nextPrimitive(NumberDef.class).intValue();
+		ClassInstance callback = arguments.nextFunction();
+		Interpreter interpreter = arguments.getInterpreter();
+		int startX = Math.min(ax, bx);
+		int endX = Math.max(ax, bx);
+		int startY = Math.min(ay, by);
+		int endY = Math.max(ay, by);
+		int startZ = Math.min(az, bz);
+		int endZ = Math.max(az, bz);
+
+		int volume = (endX - startX) * (endY - startY) * (endZ - startZ);
+		int completed = 0;
+
+		ScriptUtils.showProgressBar(interpreter);
+		for (int x = startX; x <= endX; x++) {
+			ClassInstance xInstance = interpreter.create(NumberDef.class, (double) x);
+			for (int y = startY; y <= endY; y++) {
+				ClassInstance yInstance = interpreter.create(NumberDef.class, (double) y);
+				for (int z = startZ; z <= endZ; z++, completed++) {
+					ClassInstance zInstance = interpreter.create(NumberDef.class, (double) z);
+					interpreter.call(callback, List.of(xInstance, yInstance, zInstance), Trace.getINTERNAL());
```

Comment:
lambda needs 3 vector inputs: local, absolute and normalized coordinates

Reply:
or at least absolute and normalized

Reply:
Then you need to implement the vector class in Minecraft first.
I cannot pass vectors into this function without having the vector class.